### DISCUSSION TITLE
Firebase C++ SDK with dart:ffi

### DISCUSSION THREAD

**Discussion by lukaszciastko** [firebase/flutterfire#5172] [Feature request]
Hi! With dar:ffi becoming stable, are there any plans to upgrade Flutter Fire to use the native C++ libraries and skip the Java/Objective-C bridge for better performance? Right now, Flutter Fire doesn't support Windows or Linux, but, with C++, I believe this would be possible (although Firebase desktop support is in beta and the number of available features are limited).

**Comment by thegrxp** [comment]
Would be amazing to be able to use Firebase for Windows and Linux as well. For my use case there is no point in having a MacOs app if it's not available on other desktop platforms

**Comment by josemanuelp** [comment]
That would be amazing!

**Comment by TimWhiting** [comment]
Is there anyone interested in collaborating on something like this or is there already an attempt out there doing this? I know there are several attempts at doing Firebase in pure dart - which I'm also interested in: however, those projects are limited in features, interoperability, and support. Using the platform interfaces seems like a big win for interoperability / features / support (since you can still use the functional features on the platforms that support them while you are waiting on support for the other platforms).
